pkg/pool-linear/contracts/test/MockMaliciousEulerToken.sol

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+
+// has many ERC20 and additional functions
+import "@balancer-labs/v2-solidity-utils/contracts/test/TestToken.sol";
+import "@balancer-labs/v2-pool-utils/contracts/test/MaliciousQueryReverter.sol";
+
+
+contract MockMaliciousEulerToken is TestToken, MaliciousQueryReverter{
```

Comment:
I would just call this `MockEulerToken` without the `Malicious` keyword; otherwise every mock token we create for Linear Pools would include this keyword and it would be redundant. Blabs, for example, has `MockAaveLendingPool` which includes malicious behavior.